.github/workflows/build.yml

```
@@ -51,12 +51,13 @@ jobs:
 
     - name: Release Marker
       if: ${{ github.event.release }}
-      run: |
-        touch src/meltano/core/tracking/.release_marker
+      # The release marker differentiates installations 'in the wild' versus internal dev builds and tests
+      run: touch src/meltano/core/tracking/.release_marker
 
     - name: Build distribution
       run: |
-        make sdist_public
+        poetry run scripts/alembic_freeze.py
+        poetry build
```

@WillDaSilva - In 

- #6145

@edgarrmondragon and I found that the `poetry build` process worked for the wheel but did not properly execute `build.py` when running for `sdist`. Discussion thread here: https://github.com/meltano/meltano/pull/6145/files#r897501873

Can we simply run `build.py` manually as in the suggestion below in order to mitigate that discovered issue?

```suggestion
        poetry run scripts/alembic_freeze.py
        poetry run scripts/build.py  # needed for sdist due to https://github.com/python-poetry/poetry/issues/2740#issuecomment-860744196
        poetry build
```

@edgarrmondragon - Any thoughts? I'd rather not remove `sdist` distributions entirely as part of this particular PR, but that's something we also discussed as a future option.

---

The `sdist` ought to contain the unbuilt source of the repository. It comes with `build.py` within it, and when it is installed, `build.py` is run, as the first step of installing an `sdist` is building a wheel out of it.

If we can accept that anyone who is installing an `sdist` should have `yarn` installed so that they can build the Meltano UI, then the fix here will be to ensure the `sdist` contains the unbuilt Javascript code in `src/webapp` in addition to `src/meltano`.

---

Want to add I really like these updates:

> We can probably actually get rid of scripts/bundle_ui.py, and just call build.py instead, which does the same thing, but is automatically run by poetry when you build a wheel. Anyone who wishes to build & bundle the UI manually can simply manually call build.py. Plus now we can have it as a Python file that just about everyone here will understand, rather than a bash script that might be intimidating to some.

The rename to `build.py` addresses some of my concerns around docs, since this is self-documenting according to conventions many would reasonably expect. A paired `clean.py` also would serve a self-documenting function, in that it would remove those files that should not remain in the tree, and be self-documenting of those files without expecting too much inference from developers/contributors.

---

> If we can accept that anyone who is installing an `sdist` should have `yarn` installed so that they can build the Meltano UI, then the fix here will be to ensure the `sdist` contains the unbuilt Javascript code in `src/webapp` in addition to `src/meltano`.

@WillDaSilva Problem is we've been shipping the `sdist` with the built UI assets, so not including them would be a breaking change for anyone relying on the `sdist`.

---

@edgarrmondragon True, but is it reasonable for anyone to specifically depend on the latest version of an `sdist` not requiring anything special to build it? It's common for `sdist`s to have many system level dependencies (often on C libraries).

The only 3 categories of users I know who depend specifically on an `sdist` are:
- Those who want to build it themselves (e.g. to build it into a Conda package instead of a wheel).
- Those who want to audit it or its build.
- People who haven't installed the `wheel` package.

The first two categories of users who depend on an `sdist` are generally fine dealing with the difficulties around building package with system-level dependencies. The third is rare, and ought to install the `wheel` package.

---

@WillDaSilva 

> but is it reasonable for anyone to specifically depend on the latest version of an `sdist` not requiring anything special to build it? It's common for `sdist`s to have many system level dependencies (often on C libraries).

Certainly not reasonable, and you're right in that it's how numpy and the like do things.

> The only 3 categories...

Since we publish a "pure Python" wheel, no user installing from PyPI should ever be forced to install from `sdist` when using pip on a niche platform, right?

> Another option to consider is that we could containerize the JS build process. That would introduce a dependency on Docker for any who wish to build the `sdist`, but that's a much simpler dependency (and less likely to change over time) than requiring specific versions of Node, Yarn, and the various C libraries they depend on.

I'm not a fan of that. I think documenting the required versions of node, yarn, etc. is reasonable enough and AFAIK can even be explicit requirements set in `package.json`.

cc @aaronsteers I think there's a good case for publishing a _raw_ `sdist` without built UI assets, and that would simplify our workflows a bit (see https://github.com/meltano/meltano/issues/6566#issuecomment-1205649134)

---

> Since we publish a "pure Python" wheel, no user installing from PyPI should ever be forced to install from sdist when using pip on a niche platform, right?

@edgarrmondragon Correct, so long as they have `wheel` installed, which is generally true. We can ensure it by adding `wheel` to our `build-system` in `pyproject.toml`, which is currently:

```toml
[build-system]
requires = ["poetry-core>=1.0.0"]
build-backend = "poetry.core.masonry.api"
```

> I think documenting the required versions of node, yarn, etc. is reasonable enough and AFAIK can even be explicit requirements set in package.json.

I'm okay with that, but it's worth noting that this requires the user either change their globally installed version of Node/Yarn, learn how to use a tool like NVM, or create a temporary environment (e.g. within an interactive Docker container or VM) in which to build Meltano. The requirement "have Docker/Podman installed" is much more simple than "have version X of Node and version Y of Yarn installed".